### ISSUE TITLE
[false positive] `These two ifs can be merged into one` with nested ifs (?)

### ISSUE THREAD

**Issue by jrudolph** [HairyFotr/linter#9]
There are two legitimately nested `if` so not sure what this is about:

```
[warn] /home/johannes/git/telfish/spray/spray/spray-can/src/main/scala/spray/can/server/StatsSupport.scala:45: These two ifs can be merged into one.
[warn]       if (currentMoc > moc)
[warn]       ^
```

Here's the complete method:

```
@tailrec
final def adjustMaxOpenConnections(): Unit = {
  val co = connectionsOpened.get
  val cc = connectionsClosed.get
  val moc = maxOpenConnections.get
  val currentMoc = co - cc
  if (currentMoc > moc)
    if (!maxOpenConnections.compareAndSet(moc, currentMoc)) adjustMaxOpenConnections()
}
```
